<launch>
  <arg name="fps_A" default="25" />

  <node name="miivii_gmsl_ros_A" pkg="miivii_gmsl_ros" type="miivii_gmsl_ros_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="1280" />
    <param name="image_height" value="720" />
    <param name="camCount" value="4" />
    <param name="enable_sync" value="1" />
    <param name="fps" value="$(arg fps_A)" />
    <param name="domain" value="" />

    <param name="camera1_topic" value="camera1" />
    <param name="camera1_name" value="front_left" />
    <param name="camera1_info_topic" value="camera1_info" />
    <param name="camera1_cali_file" value="file:///home/nvidia/workspace/robotcar_ws/src/gmsl_camera/config/front_left.yaml" />
    <param name="camera1_frame_id" value="camera_front_left" />

    <param name="camera2_topic" value="camera2" />
    <param name="camera2_name" value="front_right" />
    <param name="camera2_info_topic" value="camera2_info" />
    <param name="camera2_cali_file" value="file:///home/nvidia/workspace/robotcar_ws/src/gmsl_camera/config/front_right.yaml" />
    <param name="camera2_frame_id" value="camera_front_right" />

    <param name="camera3_topic" value="camera3" />
    <param name="camera3_name" value="front_mid" />
    <param name="camera3_info_topic" value="camera3_info" />
    <param name="camera3_cali_file" value="file:///home/nvidia/workspace/robotcar_ws/src/gmsl_camera/config/front_mid.yaml" />
    <param name="camera3_frame_id" value="camera_front_mid" />

    <param name="camera4_topic" value="camera4" />
    <param name="camera4_name" value="rear" />
    <param name="camera4_info_topic" value="camera4_info" />
    <param name="camera4_cali_file" value="file:///home/nvidia/workspace/robotcar_ws/src/gmsl_camera/config/rear.yaml" />
    <param name="camera4_frame_id" value="camera_rear" />
  </node>


  <!--group ns = "stereo">
      <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc" output = "screen">
        <remap from="/stereo/left/image_raw" to="/miivii_gmsl_ros_A/camera1"/>
        <remap from="/stereo/right/image_raw" to="/miivii_gmsl_ros_A/camera2"/>
        <remap from="/stereo/left/camera_info" to="/miivii_gmsl_ros_A/front_left/camera_info"/>
        <remap from="/stereo/right/camera_info" to="/miivii_gmsl_ros_A/front_right/camera_info"/>
      </node>
  </group-->

</launch>
